From 067e5119d37545c22ca0a0d8327e2b55bb8ce45c Mon Sep 17 00:00:00 2001
From: Vincent ABRIOU <vincent.abriou@st.com>
Date: Thu, 3 Sep 2020 13:02:43 +0200
Subject: [PATCH 1/1] fix cxxopts and ghc cross compilation issue

While cross compiling using openembedded yocto environment,
CXXOPTS_INCLUDE and GHC_INCLUDE variables remains empty because the search
path point to the openembedded sysroot instead of the current directory.
To avoid this situation, NO_CMAKE_FIND_ROOT_PATH must be added to the
find_path call.

Signed-off-by: Vincent ABRIOU <vincent.abriou@st.com>
Change-Id: I3ba17a300d5ca7fbc58b6ec5c4eb15267dcee228
---
 cmake/GlobalConfig.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/GlobalConfig.cmake b/cmake/GlobalConfig.cmake
index cc44efdd..1db0025d 100644
--- a/cmake/GlobalConfig.cmake
+++ b/cmake/GlobalConfig.cmake
@@ -142,13 +142,13 @@ endif()
 
 if (NOT BUILD_PIPE_ONLY)
   # cxxopts (Alternative to boost::program_options)
-  find_path(CXXOPTS_INCLUDE cxxopts/cxxopts.hpp PATHS third-party)
+  find_path(CXXOPTS_INCLUDE cxxopts/cxxopts.hpp PATHS third-party NO_CMAKE_FIND_ROOT_PATH)
   include_directories(SYSTEM "${CXXOPTS_INCLUDE}")
 endif()
 
 if (NOT BUILD_PIPE_ONLY)
   # ghc (Alternative to boost::filesystem)
-  find_path(GHC_INCLUDE ghc/filesystem.hpp PATHS third-party)
+  find_path(GHC_INCLUDE ghc/filesystem.hpp PATHS third-party NO_CMAKE_FIND_ROOT_PATH)
   include_directories(SYSTEM "${GHC_INCLUDE}")
 endif()
 
-- 
2.17.1

